test/metabase/pulse/render/body_test.clj

```diff
@@ -621,6 +622,50 @@
                             doc)]
             (is (not (render-error? pulse-body)))))))))
 
+(def ^:private funnel-rows
+  [["cart" 1500]
+   ["checkout" 450]
+   ["homepage" 10000]
+   ["product_page" 5000]
+   ["purchase" 225]])
+
+(tx/defdataset funnel-data
+  [["stages"
+    [{:field-name "stage", :base-type :type/Text}
+     {:field-name "count", :base-type :type/Quantity}]
+    funnel-rows]])
+
+(deftest render-funnel-text-row-labels-test-2
```

Comment:
needs a `^:synchronized`, no?

Reply:
This is a question for Tim, but I assume that `mt/dataset` is the non-threadsafe part? I see for h2 at least it'll clobber an existing table for the same dataset.

Since this test is the only one using that dataset, and thus the corresponding database, it looks like this can actually be `:parallel`? I guess that's brittle because it'd be easy for another test to use the same dataset, or at least a binding with the same name, in the same namespace (which determines the database name).